app/client/src/components/shared/MapMouseEvents.js

```diff
@@ -1,16 +1,19 @@
 import { useCallback, useContext, useEffect, useRef, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
+import * as reactiveUtils from '@arcgis/core/core/reactiveUtils';
```

Comment:
Nothing to change here. Awesome! This was just added with the latest version of ArcGIS JS API. Glad to see that you are using the latest features.

It sounds like eventually the `watchUtils` library will be deprecated, so at some point we will need to completely switch over to `reactiveUtils`.